walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -37,89 +37,102 @@
 
-            // If there is at least one required or optional attendee in the given event, this event should affect the returned time range.
-            if (!Collections.disjoint(attendeesRequestList, event.getAttendees())) {
+            // If there is at least one requested attendee in the given event, this event should affect the returned time range.
+            if (!Collections.disjoint(attendeesRequest, event.getAttendees())) {
                 conflicts.add(event.getWhen());
             }
         }
 
         return findAvailableTimeSlot(conflicts, durationRequest);
     }
 
-    public List<TimeRange> disconsiderOptionalAttendees(Collection<Event> events, Collection<String> attendeesRequest, long durationRequest) {
-        List<TimeRange> conflicts = new ArrayList<TimeRange>();
-        Iterator<Event> eventsIterator = events.iterator();
-        while (eventsIterator.hasNext()) {
-            Event event = eventsIterator.next();
-
-            // If there is at least one required attendee in the given event, this event should affect the returned time range.
-            if (!Collections.disjoint(attendeesRequest, event.getAttendees())) {
-                conflicts.add(event.getWhen());
-            }
-        }
+    public List<TimeRange> considerAttendees(Collection<Event> events, Collection<String> attendeesRequest, Collection<String> attendeesOptional, long durationRequest) {
+        // Add optional attendees to the requested attendees. 
+        ArrayList<String> attendeesRequestList = new ArrayList<String>(attendeesRequest);
+        ArrayList<String> attendeesOptionalList = new ArrayList<String>(attendeesOptional);
+        attendeesRequestList.addAll(attendeesOptionalList);
 
-        return findAvailableTimeSlot(conflicts, durationRequest);
+        return considerAttendees(events, attendeesRequestList, durationRequest);
     }
 
+    // Returns a list of available time ranges for the requested event, given its duration and a list of time ranges for any conflicting events.
     public List<TimeRange> findAvailableTimeSlot(List<TimeRange> conflicts, long durationRequest) {
         Collections.sort(conflicts, TimeRange.ORDER_BY_START);
 
         List<TimeRange> available = new ArrayList<TimeRange>();
         int start = TimeRange.START_OF_DAY;
         int end = TimeRange.END_OF_DAY;
         int first = 1;
-        for (TimeRange when : conflicts) {
+        for (TimeRange conflict : conflicts) {
+            // If this is the first conflict seen, then set the start and end variables as the start and end of that conflict.
             if (first == 1) {
                 first = 0;
-                start = when.start();
-                end = when.end();
+                start = conflict.start();
+                end = conflict.end();
+
+                // If there is enough time for the requested event to happen before the first conflic, then add the time range from the start
```

Comment:
conflict

Reply:
Sorry, I didn't quite get what the conflict is here.

Reply:
Oh, i meant that you typoed the word "conflict" in your comment here.